mempool/src/shared_mempool.rs

```diff
@@ -173,14 +198,42 @@ async fn sync_with_peers<'a>(
     }
 }
 
-/// used to validate incoming transactions and add them to local Mempool
+fn convert_txn_from_proto(txn_proto: SignedTransactionProto) -> Option<SignedTransaction> {
+    match SignedTransaction::try_from(txn_proto.clone()) {
+        Ok(txn) => Some(txn),
+        Err(e) => {
+            security_log(SecurityEvent::InvalidTransactionMP)
+                .error(&e)
+                .data(&txn_proto)
+                .log();
+            None
+        }
+    }
+}
+
+/// submits a list of SignedTransaction to the local mempool,
+/// and returns an AdmissionControlStatus if `is_ac_endpoint_submission`
+/// if `is_ac_endpoint_submission`, we expect a single transaction to be submitted
 async fn process_incoming_transactions<V>(
     smp: SharedMempool<V>,
-    peer_id: PeerId,
+    peer_id: Option<PeerId>,
     transactions: Vec<SignedTransaction>,
-) where
+) -> Option<Status>
+where
     V: TransactionValidation,
 {
+    let (peer_id, is_ac_endpoint_submission, timeline_state) = match peer_id {
+        Some(peer_id) => {
+            if smp.validator_peers.contains(&peer_id) {
```

Comment:
I don't think you really need this DS or this DS is enough. Imagine that you'd want in a future to implement similar policy for VFN network.
Can we instead check that message come from downstream network?